### ISSUE TITLE
Dynamic imports throw when combined with `composite: true`

### ISSUE THREAD

**Issue by alejandroclaro** [wessberg/rollup-plugin-ts#83]
When code splitting is configured the `tsc` fail with the following error:

``` text
[!] (plugin Typescript) TS6307: File '.../my-package/src/bar.ts' is not listed within the file list of project '.../my-package/tsconfig.json'. Projects must list all files or use an 'include' pattern.

src/index.ts (4:10)
4   import('./bar').catch((error) => {
           ~~~~~~~
error Command failed with exit code 1.
```

There is a project prepared to reproduce the issue:

[https://github.com/alejandroclaro/rollup-plugin-ts-bug-83.git](https://github.com/alejandroclaro/rollup-plugin-ts-bug-83.git)

This i the rollup configuration:

``` javascript
const Path = require('path');

const commonjs = require('@rollup/plugin-commonjs');
const resolve = require('@rollup/plugin-node-resolve');
const typescript = require('@wessberg/rollup-plugin-ts');

const config = {
  treeshake: false,
  plugins: [
    resolve({ 'browser': false, 'preferBuiltins': true }),
    commonjs(),
    typescript({
      'tsconfig': (resolvedConfig) => {
        const config = { ...resolvedConfig };
        config.declarationDir = Path.join(__dirname, 'dist', 'types');
        return config;
      }
    })
  ],
  external: [
  ],
  input: './src/index.ts',
  output: [
    {
      'dir': 'dist',
      'entryFileNames': 'dist/rollup-plugin-ts-bug.cjs.js',
      'format': 'cjs'
    }
  ]
}

export default config;
```
and the code that produces the error:

``` typescript
const processInfo = process as { readonly type?: string; };

if (processInfo.type === 'renderer' || processInfo.type === 'worker') {
  import('@src/bar').catch((error) => {
   console.log(error.message);
  });
}

export * from '@src/foo';
```

**Comment by wessberg** [comment]
Hey there. Sorry for the late reply. I've narrowed this down to be related to the combination of dynamic imports and the `composite: true` option inside a `tsconfig`. I'm currently investigating why it happens and how to work around it.

**Comment by wessberg** [comment]
This was due to a special case in which a certain cache wasn't being correctly invalidated. It has been fixed as part of [v1.2.31](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.2.31). Nice catch!
